other/web-js/client/src/client-api.js

```diff
@@ -0,0 +1,212 @@
+import jwtDecode from "jwt-decode";
+
+/**
+ * Connects to a MongoDB Atlas App Services App's client API over HTTPS
+ */
+export class ClientApi {
+  static constructBaseUrl(deployment) {
+    const { deployment_model, cloud, region } = deployment;
+    if (deployment_model === "GLOBAL") {
+      return `https://realm.mongodb.com/api/client/v2.0/`;
+    }
+    if (!cloud || !region) {
+      throw new Error(
+        `Must specify a cloud provider and region for LOCAL Apps. e.g. { "cloud": "aws", "region": "us-east-1" }`
+      );
+    }
+    return `https://${region}.${cloud}.realm.mongodb.com/api/client/v2.0/`;
+  }
+
+  constructor({ appId, deployment_model, cloud, region, onAuthChange }) {
+    this.appId = appId;
+    this.credentialStorage = new CredentialStorage(appId);
+    this.currentUser = this.credentialStorage.get("currentUser");
+    this.deployment = {
+      deployment_model:
+        deployment_model ?? (cloud || region) ? "LOCAL" : "GLOBAL",
+      cloud,
+      region,
+    };
+    this.baseUrl = ClientApi.constructBaseUrl(this.deployment);
+    this.onAuthChange = onAuthChange;
+  }
+
+  // endpointUrl(path: string): string;
```

Comment:
rather than these comments showing what's going on, can you refactor them using JSDoc?

here for example, it'd be:

```suggestion
  /**
  * @param {string} path
  * @returns {string}
  */
```